aptos-move/e2e-testsuite/src/tests/failed_transaction_tests.rs

```diff
@@ -1,118 +0,0 @@
-// Copyright © Aptos Foundation
-// Parts of the project are originally copyright © Meta Platforms, Inc.
-// SPDX-License-Identifier: Apache-2.0
-
-use aptos_gas_meter::{StandardGasAlgebra, StandardGasMeter};
-use aptos_gas_schedule::{AptosGasParameters, LATEST_GAS_FEATURE_VERSION};
-use aptos_language_e2e_tests::{common_transactions::peer_to_peer_txn, executor::FakeExecutor};
-use aptos_memory_usage_tracker::MemoryTrackedGasMeter;
-use aptos_types::{
-    state_store::{state_key::StateKey, TStateView},
-    transaction::ExecutionStatus,
-    vm_status::{StatusCode, VMStatus},
-    write_set::WriteOp,
-};
-use aptos_vm::{data_cache::AsMoveResolver, transaction_metadata::TransactionMetadata, AptosVM};
-use aptos_vm_logging::log_schema::AdapterLogSchema;
-use aptos_vm_types::storage::StorageGasParameters;
-use claims::assert_some;
-use move_core_types::vm_status::StatusCode::TYPE_MISMATCH;
-
-#[test]
-fn failed_transaction_cleanup_test() {
```

Comment:
why the test deleted?

Reply:
Ah, so I realised this test was unreachable in the current flow (we check if status's is discard, otherwise call the function which is being tested here for ... discards). But I unified the handling there so the test actually is reachable now, forgot to revive🤦‍♂️

Reply:
In any case, will add a test, but my plan is to start adding a few tests for the error flows to ensure we have a good coverage there

Reply:
Rewrote the test using failpoints, see `vm.rs`. I find testing VM hard because we have no abstractions so it is hard to mock it, reducing the dependency on concrete types should make it easier, what do you think?

Reply:
hmm, why is the test unreachable? you mean the TYPE_MISMATCH and UNKNOWN_INVARIANT_VIOLATION_ERROR are discarded? I thought we charge for them no? and the new test doesn't cover them

Reply:
Well, they are not charged but kept. This test is seems wrong: we initialise `TransactionMetadata` with 100_000 max gas amount, but `gas_meter` with a balance of 10_000🙃.  But for execution, we use 
```
let balance = TransactionMetadata::new(txn).max_gas_amount();
```
So `failed_transaction_cleanup_test` cannot charge anything here, and its crazy that we have an assertion here `assert_eq!(out1.gas_used(), 90_000);`. 

What should be checked, is a kept status.

Reply:
Updated the test re-using the logic from this test, but setting the gas meter correctly.